# This patch makes Ruby use the versions of libffi and OpenSSL from our build system instead of trying to look for libffi and OpenSSL using pkg-config.

--- a/ext/fiddle/extconf.rb
+++ b/ext/fiddle/extconf.rb
@@ -47,7 +47,7 @@ end
 libffi_version = nil
 have_libffi = false
 bundle = enable_config('bundled-libffi')
-unless bundle
+unless bundle; ffi_header = 'ffi.h'; have_ffi_header = true; File.open('mkxp-libffi-version-test.c', 'w') do |f| f.write("#include <#{ffi_header}>\nFFI_VERSION_STRING") end; libffi_version_command = "#{RbConfig::CONFIG['CPP']} #{RbConfig::CONFIG['CFLAGS']} mkxp-libffi-version-test.c"; $stderr.puts(libffi_version_command); libffi_version = `#{libffi_version_command}`.lines[-1].split('"')[1]; $libs = append_library($libs, 'ffi'); have_libffi = true; end; if false
   dir_config 'libffi'
 
   if pkg_config("libffi")
--- a/ext/openssl/extconf.rb
+++ b/ext/openssl/extconf.rb
@@ -49,7 +49,7 @@ def find_openssl_library
   end
 
   return false unless have_header("openssl/ssl.h")
-
+  have_library('atomic'); $libs = append_library($libs, 'crypto'); $libs = append_library($libs, 'ssl'); return true
   ret = have_library("crypto", "CRYPTO_malloc") &&
     have_library("ssl", "SSL_new")
   return ret if ret
@@ -92,7 +92,7 @@ def find_openssl_library
 end
 
 Logging::message "=== Checking for required stuff... ===\n"
-pkg_config_found = !dir_config_given && pkg_config("openssl") && have_header("openssl/ssl.h")
+pkg_config_found = false
 
 if !pkg_config_found && !find_openssl_library
   Logging::message "=== Checking for required stuff failed. ===\n"
